src/main/kotlin/org/veupathdb/service/eda/compute/EDA.kt

```diff
@@ -0,0 +1,182 @@
+
+  /**
+   * Fetches the [APIStudyDetail] information from the EDA Subsetting Service
+   * for the given [studyID] if such a study exists.
+   *
+   * @param studyID ID of the study whose metadata should be retrieved.
+   *
+   * @param auth Auth header sent in with the job HTTP request.
+   *
+   * @return An [Optional] that will wrap the `APIStudyDetail` returned from the
+   * EDA Subsetting Service, if such a study exists.  If the target study was
+   * not found, the returned `Optional` will be empty.
+   */
+  @JvmStatic
+  fun getAPIStudyDetail(studyID: String, auth: TwoTuple<String, String>): Optional<APIStudyDetail> =
+    EdaSubsettingClient(ServiceOptions.edaSubsettingHost, auth).getStudy(studyID)
+
+  /**
+   * Fetches the [APIStudyDetail] information from the EDA Subsetting Service
+   * for the given [studyID], throwing an exception if no such study exists.
+   *
+   * @param studyID ID of the study whose metadata should be retrieved.
+   *
+   * @param auth Auth header sent in with the job HTTP request.
+   *
+   * @param err Optional exception provider that will be used to get the
+   * exception that will be thrown if the target study does not exist.
+   *
+   * @return The `APIStudyDetail` information returned from the EDA Subsetting
+   * Service.
+   *
+   * @throws Exception Throws the exception returned by the given exception
+   * provider ([err]).
+   */
+  @JvmStatic
+  @JvmOverloads
+  fun requireAPIStudyDetail(
+    studyID: String,
+    auth: TwoTuple<String, String>,
+    err: (studyID: String) -> Exception = this::noStudyDetail
+  ): APIStudyDetail = getAPIStudyDetail(studyID, auth).orElseThrow { err(studyID) }
+
+  /**
+   * Fetches tabular study data from the EDA Merge Service for the given params.
+   *
+   * @param refMeta TODO: What is this?
+   *
+   * @param filters TODO: What is this?
+   *
+   * @param spec TODO: What is this?
+   *
+   * @param auth Auth header sent in with the job HTTP request.
+   *
+   * @return An [InputStream] over the tabular data returned from the EDA Merge
+   * Service.
+   */
+  @JvmStatic
+  fun getMergeData(
+    refMeta: ReferenceMetadata,
+    filters: List<APIFilter>,
+    spec: StreamSpec,
+    auth: TwoTuple<String, String>
+  ): InputStream =
+    EdaMergingClient(ServiceOptions.edaMergeHost, auth)
+      .getTabularDataStream(refMeta, filters, spec).inputStream
+
+
+  /**
+   * Submits a new compute job to the queue.
+   *
+   * @param plugin Plugin details.
+   *
+   * @param payload HTTP request containing the compute config.
+   *
+   * @param auth: Auth header sent in with the job HTTP request.
+   *
+   * @return A response describing the job that was created.
+   */
+  @JvmStatic
+  fun <R : ComputeRequestBase> getOrSubmitComputeJob(
+    plugin:  PluginMeta<R>,
+    payload: R,
+    auth:    TwoTuple<String, String>,
+  ): JobResponse {
+    // Serialize the http request to json
+    val serial = Json.convert(payload)
```

Comment:
does the payload include any JSON arrays?  how are u dealing with getting their elements into a cannonical order?

Reply:
The payload may or may not include JSON arrays dependent on the plugin configuration.

Right now, since the plugin configuration is effectively a black box to the compute service, it assumes the client or requester submits the data with the correct relevant sorting.  I intentionally did not attempt to sort the data because I cannot safely make a guess as to what arrays have an ordering that is important vs what arrays can have their elements appear in any order.